### PR TITLE
Solved minor bug with MLFlow logger

### LINKED ISSUE #16411: Typo breaks mlflow logger in 1.9.0
### Bug description

This [line](https://github.com/Lightning-AI/lightning/blob/898a236681def29aec9ce77e02fc5a15dd5d9cb3/src/pytorch_lightning/loggers/mlflow.py#L250) of code appears to be broken in 1.9.0 as it erroneously passes the value of the parameter where it should be providing the name:

```py
params_list.append(Param(key=v, value=v))
```

This should instead likely be:

```py
params_list.append(Param(key=k, value=v))
```

### How to reproduce the bug

```
from collections import OrderedDict
from lightning.pytorch.loggers import MLFlowLogger

params = OrderedDict()
params["a"] = 1

MLFlowLogger().log_hyperparams(params)
```

### Error messages and logs

```
TypeError: expected string or bytes-like object
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0): 1.9.0
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): source
#- Running environment of LightningApp (e.g. local, cloud): local
```

</details>

### More info

_No response_

cc @borda

### PR DESCRIPTION
**Pull request by BrianPulfer**
Resolves #16411 the minor issue #16411 (typo) from

`params_list.append(Param(key=v, value=v))`

to 

`params_list.append(Param(key=k, value=v))`

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by senarvi** [comment]
For me this fixes MLflow logging.

But now that I look at the code, what do you think of replacing the entire loop with this:

`params_list = [Param(key=k, value=str(v)[:250]) for k, v in params.items()]`

i.e. instead of not logging long string values, we would log the first 250 characters?

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
@senarvi That sounds good to me. Feel free to send a PR with this improvement. Fewer warnings are always better if it can be avoided :)

File: src/lightning_fabric/CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
--
+- Fixed an issue with `MLFlowLogger` logging the wrong keys with `.log_hyperparams()` ([#16418](https://github.com/Lightning-AI/lightning/pull/16418))
```
Comment (carmocca) [inline src/lightning_fabric/CHANGELOG.md:32]: Wrong CHANGELOG @awaelchli

**Review by carmocca — APPROVED** [review]

**Comment by schmidt-jake** [comment]
@senarvi thanks for catching this!

As for your suggestion, two thoughts:
1. Truncating keys to 250 characters may result in duplicate keys — thoughts on how to handle this?
2. [MLFlow version 1.28](https://github.com/mlflow/mlflow/releases/tag/v1.28.0) increased the max param length to 500 — would be nice to take advantage of this

**Pull request merged**